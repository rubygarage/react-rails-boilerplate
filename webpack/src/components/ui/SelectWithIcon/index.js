import React from 'react'
import PropTypes from 'prop-types'
import ReactSelect from 'react-select'
import SelectOption from 'components/ui/SelectOption'
import SelectValue from 'components/ui/SelectValue'

export function SelectWithIcon({ className, onChange, options, placeholder, value }) {
  return (
    <ReactSelect
      className={className}
      name="select-with-icon"
      onChange={onChange}
      optionComponent={SelectOption}
      options={options}
      placeholder={placeholder}
      searchable={false}
      value={value}
      valueComponent={SelectValue}
    />
  )
}

SelectWithIcon.propTypes = {
  className: PropTypes.string,
  onChange: PropTypes.func,
  options: PropTypes.arrayOf(PropTypes.shape({
    icon: PropTypes.string,
    label: PropTypes.string,
    value: PropTypes.string
  })),
  placeholder: PropTypes.string,
  value: PropTypes.string
}

export default SelectWithIcon
