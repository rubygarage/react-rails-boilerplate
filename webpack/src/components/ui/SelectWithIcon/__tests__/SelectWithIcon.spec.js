import React from 'react'
import { shallow } from 'enzyme'
import SelectWithIcon from 'components/ui/SelectWithIcon'

describe('<SelectWithIcon />', () => {
  const defaultProps = {
    value: 'Cardable::AudioCard',
    options: [
      { value: 'Cardable::AudioCard', label: 'Audio', icon: 'icon-audio-circle' },
      { value: 'Cardable::VideoCard', label: 'Video', icon: 'icon-play-circle' }
    ],
    placeholder: 'Select card type'
  }

  it('renders select with icon', () => {
    const component = shallow(
      <SelectWithIcon {...defaultProps} />
    )

    expect(component).toMatchSnapshot()
  })
})
