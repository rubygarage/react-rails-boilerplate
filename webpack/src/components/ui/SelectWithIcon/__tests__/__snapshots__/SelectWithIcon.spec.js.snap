// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectWithIcon /> renders select with icon 1`] = `
<Select
  addLabelText="Add \\"{label}\\"?"
  arrowRenderer={[Function]}
  autosize={true}
  backspaceRemoves={true}
  backspaceToRemoveMessage="Press backspace to remove {label}"
  clearAllText="Clear all"
  clearRenderer={[Function]}
  clearValueText="Clear value"
  clearable={true}
  deleteRemoves={true}
  delimiter=","
  disabled={false}
  escapeClearsValue={true}
  filterOptions={[Function]}
  ignoreAccents={true}
  ignoreCase={true}
  inputProps={Object {}}
  isLoading={false}
  joinValues={false}
  labelKey="label"
  matchPos="any"
  matchProp="any"
  menuBuffer={0}
  menuRenderer={[Function]}
  multi={false}
  name="select-with-icon"
  noResultsText="No results found"
  onBlurResetsInput={true}
  onCloseResetsInput={true}
  optionComponent={[Function]}
  options={
    Array [
      Object {
        "icon": "icon-audio-circle",
        "label": "Audio",
        "value": "Cardable::AudioCard",
      },
      Object {
        "icon": "icon-play-circle",
        "label": "Video",
        "value": "Cardable::VideoCard",
      },
    ]
  }
  pageSize={5}
  placeholder="Select card type"
  required={false}
  scrollMenuIntoView={true}
  searchable={false}
  simpleValue={false}
  tabSelectsValue={true}
  value="Cardable::AudioCard"
  valueComponent={[Function]}
  valueKey="value"
/>
`;
